src/Service/Resolvers/BaseQueryStructure.cs

```diff
@@ -109,11 +111,19 @@ public class BaseQueryStructure
         ///  Add parameter to Parameters and return the name associated with it
         /// </summary>
         /// <param name="value">Value to be assigned to parameter, which can be null for nullable columns.</param>
-        public string MakeParamWithValue(object? value)
+        public virtual string MakeParamWithValue(object? value, string? paramName = null)
         {
-            string paramName = $"{PARAM_NAME_PREFIX}param{Counter.Next()}";
-            Parameters.Add(paramName, value);
-            return paramName;
+            string encodedParamName = $"{PARAM_NAME_PREFIX}param{Counter.Next()}";
+            if (!string.IsNullOrEmpty(paramName))
+            {
+                Parameters.Add(encodedParamName, new(value, GetUnderlyingSourceDefinition().GetDbTypeForParam(paramName)));
+            }
+            else
+            {
+                Parameters.Add(encodedParamName, new(value, null));
```

Comment:
Why is DbType null if the paramName is null or empty? 
DbType should always have some value right? is this because we dont have same implementation for MySql/PostgreSql - if thats the case, I think its better to have different implementations of MakeParamWithValue for different dbs instead of requiring nullable DbType

Reply:
What was the resolution to this comment , since it was marked closed?